src/vs/workbench/parts/stats/node/workspaceStats.ts

```diff
@@ -343,36 +399,60 @@ export class WorkspaceStats implements IWorkbenchContribution {
-					return this.fileService.resolveContent(uri, { acceptTextOnly: true }).then(content => {
-						try {
-							const packageJsonContents = JSON.parse(content.value);
-							if (packageJsonContents['dependencies']) {
-								for (let module of ModulesToLookFor) {
-									if ('react-native' === module) {
-										if (packageJsonContents['dependencies'][module]) {
-											tags['workspace.reactNative'] = true;
-										}
-									} else {
-										if (packageJsonContents['dependencies'][module]) {
-											tags['workspace.npm.' + module] = true;
-										}
-									}
+			function getFilePromises(filename, fileService, contentHandler): TPromise<void>[] {
+				return !nameSet.has(filename) ? [] : folders.map(workspaceUri => {
+					const uri = workspaceUri.with({ path: `${workspaceUri.path !== '/' ? workspaceUri.path : ''}/${filename}` });
+					return fileService.resolveFile(uri).then(() => {
+						return fileService.resolveContent(uri, { acceptTextOnly: true }).then(contentHandler);
+					}, err => {
+						// Ignore missing file
+					});
+				});
+			}
+
+			let dependencyFilePromises;
+
+			const requirementsTxtPromises = getFilePromises('requirements.txt', this.fileService, content => {
+				console.log(content.value);
+				let dependencies: string[] = content.value.split('\n');
+				tags['workspace.py.any-azure'] = WorkspaceStats.searchArray(dependencies, /azure/i);
+				for (let dependency of dependencies) {
+					let packageName = dependency.split('==')[0];
+					if (PyModulesToLookFor.indexOf(packageName) > -1) {
+						tags['workspace.py.' + packageName] = true;
+					}
+					// congitive services has a lot of tiny packages. eg. 'azure-cognitiveservices-search-autosuggest'
+					if (packageName.indexOf('azure-cognitiveservices') > -1) {
+						tags['workspace.py.cognitiveservices'] = true;
+					}
+				}
+			});
+
+			const packageJsonPromises = getFilePromises('package.json', this.fileService, content => {
+				try {
+					const packageJsonContents = JSON.parse(content.value);
+					if (packageJsonContents['dependencies']) {
+						for (let module of ModulesToLookFor) {
+							if ('react-native' === module) {
+								if (packageJsonContents['dependencies'][module]) {
+									tags['workspace.reactNative'] = true;
+								}
+							} else {
+								if (packageJsonContents['dependencies'][module]) {
+									tags['workspace.npm.' + module] = true;
 								}
 							}
 						}
-						catch (e) {
-							// Ignore errors when resolving file or parsing file contents
-						}
-					});
-				}, err => {
-					// Ignore missing file
-				});
+					}
+				}
+				catch (e) {
+					// Ignore errors when resolving file or parsing file contents
+				}
 			});
 
-			return TPromise.join(packageJsonPromises).then(() => tags);
+			dependencyFilePromises = packageJsonPromises.concat(requirementsTxtPromises);
+
+			return TPromise.join(dependencyFilePromises).then(() => tags);
```

Comment:
The variable `dependencyFilePromises` is not needed. You can do `TPromise.join([...packageJsonPromises, ...requirementsTxtPromises])